### DISCUSSION TITLE
About the level building and tilemaps and allat

### DISCUSSION THREAD

**Discussion by zacnmac** [Techokami/SonicWorldsNext#112] [Q&A]
Would how to give tiles collision be more of a Godot question or more of a Worlds question? I'm aware that layers determine what layer a character or object or what have you is on, and a mask makes it so that the item knows what layer it can collide with. However, I have no idea how to actually use this information to make levels.

**Comment by DimensionWarped** [comment]
>Would how to give tiles collision be more of a Godot question or more of a Worlds question?

A little bit of both. Understanding the collision layers is important of course, but the bigger piece to this puzzle is understanding the Godot resource 'tileset' as well as the Godot object 'tilemap'. Outside of that you can get by temporarily by just giving your tileset layers 1-4 so that they have all the basic collision to act as floors/walls/ceilings. Once you want loop-de-loops, crisscrossing platforms like in Chemical Plant/Hydrocity Zone, or other wacky multilayer complexities, you'll want to get deeper into the Worlds specific elements, but you really don't actually *need* any of that stuff to make a good level (Just off the top of my head, Marble Garden, Spring Yard, Labyrinth Zone, and Scrap Brain zone don't even use these features in the first game).

Here is a brief primer of how to use Godot tilesets that should help you get started. It probably isn't going to cover collision to an elaborate level, but it should be enough to teach you the basics of making the basic map of the level. We can delve deeper as you come across things you want to make.

https://www.youtube.com/watch?v=w9QOMTBjmgg

**Reply by zacnmac** [reply]
Upon giving collision to, painting, and then placing down a few tiles for the player to stand on, as soon as I attempted to run the scene, this error came up.

![Screenshot 2023-11-26 014258](https://github.com/Techokami/SonicWorldsNext/assets/143913996/d82cea88-24e6-4b5a-b38d-8fbe73f31e8b)

**Comment by JimmyThrush** [comment]
Late reply here, but I suggest simply using the tileset that comes with the framework, rather than trying to define collision yourself. It's completely functional for almost any of your needs in typical Sonic level design.

**Reply by zacnmac** [reply]
Is there any way to overlay sprites onto that tileset? The sprites for the tilemap don't fit very well into any real level theme for obvious reasons.

**Reply by Renhoex** [reply]
You could set the tile map collision node to be invisible and use a different tilemap with all your visual tiles over it if that's what you mean

**Comment by zacnmac** [comment]
That would actually work, because I would be able to still see the color
differences between each tilemap when building but they wouldn't show up in
gameplay. yippee
